bigtable-client-core-parent/bigtable-hbase-integration-tests-common/src/test/java/com/google/cloud/bigtable/hbase/TestScan.java

```diff
@@ -305,4 +309,104 @@ public void testStartEndEquals() throws IOException {
                   .toByteArray()));
     }
   }
+
+  @Test
+  public void testColumnFamilyTimeRange() throws IOException {
+    TableName tableName = sharedTestEnv.newTestTableName();
+    String prefix = "scan_row_";
+    int rowsToWrite = 3;
+    // Initialize variables
+    sharedTestEnv.createTable(tableName);
+    Table table = getConnection().getTable(tableName);
+
+    byte[][] rowKeys = new byte[rowsToWrite][];
+    rowKeys[0] = dataHelper.randomData(prefix);
+    for (int i = 1; i < rowsToWrite; i++) {
+      rowKeys[i] = rowFollowingSameLength(rowKeys[i - 1]);
+    }
+
+    byte[][] columnFamilies = {COLUMN_FAMILY, COLUMN_FAMILY2};
+    String[][] dateRange = {
+      {"1992-04-30", "2000-01-01", "2000-12-31", "2005-06-30", "2009-12-01", "2019-02-01"},
+      {"1989-01-30", "1990-01-01", "1992-04-30", "1993-08-30", "2010-03-10", "2011-10-15"},
+    };
+
+    int numValuesPerRow = 5;
+    byte[][] quals = dataHelper.randomData("qual-", numValuesPerRow);
+    byte[][] values = dataHelper.randomData("value-", numValuesPerRow);
+
+    ArrayList<Put> puts = new ArrayList<>(rowsToWrite);
+
+    // Insert some columns
+    for (int rowIndex = 0; rowIndex < rowsToWrite; rowIndex++) {
+      Put put = new Put(rowKeys[rowIndex]);
+      for (int colFamily = 0; colFamily < columnFamilies.length; colFamily++) {
+        for (int qualifierIndex = 0; qualifierIndex < numValuesPerRow; qualifierIndex++) {
+          long time = Date.valueOf(dateRange[colFamily][qualifierIndex]).getTime();
+          put.addColumn(
+              columnFamilies[colFamily], quals[qualifierIndex], time, values[qualifierIndex]);
+        }
+      }
+      puts.add(put);
+    }
```

Comment:
Please clean up the data population. Its very hard to follow there are bunch of parallel data structures that have implicit relationships to each other:
- data ranges are coupled with qualifiers
- why do the rows have to contiguous?
-

Reply:
Also, if you break this up into smaller tests, you won't have to create so much test data

Reply:
Thanks for the suggestion, Have now break it up in two test case.